### PR TITLE
print reads on spark and tests

### PR DESCRIPTION
**Pull request by akiezun**
another small program on spark - useful for scalability testing of read/writes

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+         */
+        if (intervalArgumentCollection.intervalsSpecified()) {
+            final List<SimpleInterval> intervals =  intervalArgumentCollection.getIntervals(readsHeader.getSequenceDictionary());
+            reads = readSource.getParallelReads(bam, intervals);
+        } else {
+            reads = readSource.getParallelReads(bam, null);
+        }
+
+        try {
+            ReadsSparkSink.writeReads(ctx, output, reads, readsHeader, shardedOutput ? ReadsWriteFormat.SHARDED : ReadsWriteFormat.SINGLE);
```
Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java]: do we have any bam-reading code that understands the sharded output?

Comment (tomwhite) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java]: `ReadsSparkSource` should read sharded output, but it lacks a test. Opened https://github.com/broadinstitute/hellbender/issues/935

File: src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        } catch (final IOException e) {
+            throw new GATKException("unable to write bam: " + e);
+        }
+    }
+
+    @Override
+    protected String getProgramName() {
+        return PrintReadsSpark.class.getSimpleName();
```
Comment (tomwhite) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java]: `getClass().getSimpleName()`

Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java]: getClass is an instance method. I don't have an object of PrintReadsSpark here.

Comment (tomwhite) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java]: It's the current object (`this`). So `return getClass().getSimpleName();` should work.

Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java]: nevermind. this whole method is misplaced

Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java]: done

Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java]: yes, i thought this was in the test class.

thanks!

Comment (akiezun) [inline src/main/java/org/broadinstitute/hellbender/tools/spark/pipelines/PrintReadsSpark.java]: oops, i thought this was in the test class. thanks

**Pull request merged**